docs/static/azure-module.asciidoc

```diff
@@ -0,0 +1,458 @@
+
+In this example 'consumer_group' will be applied to each of the configured Event
+Hubs. Note that 'decorate_events' is defined in both the 'global' and per Event Hub
+configuration. The per Event Hub configuration takes precedence, and the
+global configuration is effectively ignored.
+
+["source","shell",subs="attributes"]
+-----
+modules:
+  - name: azure
+    var.elasticsearch.hosts: "localhost:9200"
+    var.kibana.host: "localhost:5601"
+    var.input.azure_event_hubs.threads: 8
+    var.input.azure_event_hubs.decorate_events: true
+    var.input.azure_event_hubs.consumer_group: logstash
+    var.input.azure_event_hubs.event_hubs:
+      - ["name",                      "event_hub_connection",      "storage_connection",                                      "initial_position", "decorate_events"]
+      - ["insights-operational-logs", "Endpoint=sb://example1...", "DefaultEndpointsProtocol=https;AccountName=example1....", "HEAD",             "true"]
+      - ["insights-metrics-pt1m",     "Endpoint=sb://example2...", "DefaultEndpointsProtocol=https;AccountName=example2....", "TAIL",             "true"]
+      - ["insights-logs-errors",      "Endpoint=sb://example3...", "DefaultEndpointsProtocol=https;AccountName=example3....", "TAIL",             "false"]
+      - ["insights-operational-logs", "Endpoint=sb://example4...", "DefaultEndpointsProtocol=https;AccountName=example4....", "HEAD",             "true"]
+-----
+
+The advanced configuration doesn't require a prefix before a per Event Hub
+configuration option. Notice the notation for the `decorate_events` option. 
+
+[[azure_best_practices]]
+===== Best practices
+
+Here are some guidelines to help you avoid data conflicts that can cause lost
+events.
+
+* **Create a {ls} consumer group.** 
+Create a new consumer group specifically for {ls}. Do not use the $default or
+any other consumer group that might already be in use. Reusing consumer groups
+among non-related consumers can cause unexpected behavior and possibly lost
+events. All {ls} instances should use the same consumer group so that they can
+work together for processing events.
+* **Avoid overwriting offset with multiple Event Hubs.** 
+The offsets (position) of the Event Hubs are stored in the configured Azure Blob
+store. The Azure Blob store uses paths like a file system to store the offsets.
+If the paths between multiple Event Hubs overlap, then the offsets may be stored
+incorrectly. 
+To avoid duplicate file paths, use the advanced configuration model and make
+sure that at least one of these options is different per Event Hub:
+** storage_connection
+** storage_container (defaults to Event Hub name if not defined)
+** consumer_group
+
+
+[[azure_config_options]]
+===== Configuration options
+
+NOTE: All Event Hubs options are common to both basic and advanced
+configurations, with the following exceptions. The basic configuration uses
+`event-hub-connections`. The the advanced configuration uses `event_hubs` and
```

Comment:
@jakelandis this seems a bit confusing from a usability standpoint. The minor syntactical difference here may throw users off. Can you elaborate on why this is the case?

Reply:
I'm happy to wait for more user feedback before we think too much into it though.

Reply:
> this seems a bit confusing from a usability standpoint

agreed. but needed. 

> The minor syntactical difference here may throw users off. Can you elaborate on why this is the case?

So... the configuration is either (pseudo configuration for illustration, not exact syntax):
```
event-hub-connections=[connection1, connection2, ...]
threads=8
```
or
```
event-hubs
|__ event-hub-connection=connection1, threads=8
|__ event-hub-connection=connection2, threads=4
```

the former supports many connections (hence the plural) the latter supports only 1 connection (hence the singular)

This is the same behavior as the input plugin for basic vs. advanced configuration. I think a user should see the discouragement of the advanced configuration (above) and not concern themselves with this ... or would copy/paste the example as a starting point (which has the correct syntax).

Reply:
Got it, then in that case there may be a typo for the advanced config option as it says `event_hubs` with an underscore rather than a dash. /cc @karenzone

Reply:
sorry, the underscores (and documentation) are correct.  my above comment incorrectly used dashes instead of underscores.

Reply:
actually... there is a typo above `event-hub-connections` should be `event_hub_connections` @karenzone sorry if misled you.  All options should use underscores not dashes.

Reply:
Corrected typo here and in azure_event_hubs plugin doc.  Called attention to plural/singular treatment for emphasis.  Let me know if this helps clarify.